src/porepy/viz/data_saving_model_mixin.py

```diff
@@ -46,19 +53,76 @@ def finalize_data_saving(self) -> None:
         if not self.suppress_export:
             self.exporter.write_pvd()
 
-    def data_to_export(self):
+    def data_to_export(self) -> list[DataInput]:
         """Return data to be exported.
 
         Return type should comply with pp.exporter.DataInput.
 
         Returns:
-            List containing all variable names.
+            List containing all (grid, name, scaled_values) tuples.
 
         """
-        var_names = [var.name for var in self.equation_system.variables]
-        # Filter out variables for interfaces of codimension 2
-        var_names = [var for var in var_names if not var.startswith("well_")]
-        return var_names
+        data = []
+        variables = self.equation_system.variables
+        for var in variables:
+            scaled_values = self.equation_system.get_variable_values([var])
+            units = var.tags["si_units"]
+            values = self.fluid.convert_units(scaled_values, units, to_si=True)
+            data.append((var.domain, var.name, values))
+
+        # Add secondary variables/derived quantities.
+        # All models are expected to have the dimension reduction methods for aperture
+        # and specific volume. More methods may be added as needed, e.g. by overriding
+        # this method:
+        #   def data_to_export(self):
+        #       data = super().data_to_export()
+        #       data.append(
+        #           (grid, "name", self._evaluate_and_scale(sd, "name", "units"))
+        #       )
+        #       return data
+        for dim in range(self.nd + 1):
+            for sd in self.mdg.subdomains(dim=dim):
+                if dim < self.nd:
+                    data.append(
+                        (sd, "aperture", self._evaluate_and_scale(sd, "aperture", "m"))
+                    )
+                data.append(
+                    (
+                        sd,
+                        "specific_volume",
+                        self._evaluate_and_scale(
+                            sd, "specific_volume", f"m^{self.nd-sd.dim}"
+                        ),
+                    )
+                )
+
+        # We combine grids and mortar grids. This is supported by the exporter, but not
+        # by the type hints in the exporter module. Hence, we ignore the type hints.
+        return data  # type: ignore[return-value]
+
+    def _evaluate_and_scale(
+        self,
+        grid: Union[pp.Grid, pp.MortarGrid],
+        method_name: str,
```

Comment:
Intuitively, I would have chosen a name as `quantity` to link to the physical nature of the object, but `method_name` seems OK as well.

Reply:
I see, but I also think it's important to convey that this name corresponds to a method (of self). I am open to changing it, though, if somebody has a brilliant suggestion (ideally covering both our conserns).

Reply:
No suggestion from my side, I think method name is least likely to cause confusion.